src/itemdef.cpp

```diff
@@ -71,6 +71,8 @@ ItemDefinition& ItemDefinition::operator=(const ItemDefinition &def)
 	stack_max = def.stack_max;
 	usable = def.usable;
 	liquids_pointable = def.liquids_pointable;
+	if (def.pointabilities)
+		pointabilities = new Pointabilities(*def.pointabilities);
```

Comment:
Why a manually managed pointer? I see no need for a pointer here. `def` is alive sufficiently long, you can have `pointables` be a raw `std::optional<Pointabilities>` object, to which you can then pass const refs around.

Reply:
I already discussed this with Desour.
Fine now I replaced all raw pointers of pointabilities with optionals, even though almost everywhere else raw pointers are used and the Minetest code style guild lines say "Do not be too C++".
(I hope it didn't break anything.)

(Note ToolCapabilities should also use std::optional.)

Reply:
> Fine now I replaced all raw pointers of pointabilities with optionals

Thanks!

> even though almost everywhere else raw pointers are used

I wouldn't use legacy code as a reference. Raw pointers have numerous downsides here.

> Do not be too C++

It seems the code style guidelines are outdated - they still mention C++ 14, while we've upgraded to C++ 17 already... Among the specific examples, I don't see anything against `std::optional` in specific, though.

> Note: ToolCapabilities should also use std::optional.

Agreed, that would be a sensible refactor for another PR.

Reply:
(The raw ptr (or `unique_ptr` preferably) would've had the advantage of less memory overhead for items without pointabilities, compared to `optional`. Idk how much of a difference it is though.)